packages/komencikit/src/kit.ts

```diff
@@ -0,0 +1,378 @@
+      TxErrorTypes.Revert,
+    ],
+    onRetry: (_args, error, attempt) => {
+      console.debug(`${TAG}/deployWallet attempt#${attempt} error: `, error)
+    },
+  })
+  public async deployWallet(
+    implementationAddress: string
+  ): Promise<Result<string, FetchError | TxError>> {
+    const resp = await this.client.exec(deployWallet({ implementationAddress }))
+    if (resp.ok) {
+      if (resp.result.status === 'deployed') {
+        return Ok(resp.result.walletAddress)
+      } else {
+        const txHash = resp.result.txHash
+        const receiptResult = await this.waitForReceipt(txHash)
+        if (receiptResult.ok) {
+          const receipt = receiptResult.result
+          if (!receipt.status) {
+            // TODO: Possible to extract reason?
+            return Err(new TxRevertError(txHash, ''))
+          }
+
+          const deployer = await this.contractKit.contracts.getMetaTransactionWalletDeployer(
+            resp.result.deployerAddress
+          )
+          const events = await deployer.getPastEvents('WalletDeployed', {
+            fromBlock: receipt.blockNumber,
+            toBlock: receipt.blockNumber,
+          })
+
+          const deployWalletLog = events.find(
+            (event) => normalizeAddressWith0x(event.returnValues.owner) === this.externalAccount
+          )
+
+          if (deployWalletLog === undefined) {
+            return Err(new TxEventNotFound(txHash, 'WalletDeployed'))
+          }
+
+          return Ok(deployWalletLog.returnValues.wallet)
+        }
+
+        return receiptResult
+      }
+    } else {
+      return resp
+    }
+  }
+
+  /**
+   * requestAttestations: uses the /v1/requestSubsidisedAttestations endpoint
+   * in order to request attestations.
+   * It constructs and passes in two meta transactions (approve, request)
+   * which are executed in batch in Komenci.
+   *
+   * @param identifier - phone number identifier
+   * @param walletAddress - MetaTransactionWallet address requesting attestations
+   * @param attestationsRequested - the number of attestations
+   * @return TransactionReceipt of the batch transaction
+   */
+  @retry({
+    tries: 3,
+    bailOnErrorTypes: [
+      FetchErrorTypes.Unauthorised,
+      FetchErrorTypes.ServiceUnavailable,
+      TxErrorTypes.Revert,
+    ],
+    onRetry: (_args, error, attempt) => {
+      console.debug(`${TAG}/requestAttestations attempt#${attempt} error: `, error)
+    },
+  })
+  public async requestAttestations(
+    identifier: string,
+    walletAddress: string,
+    attestationsRequested: number
+  ): Promise<Result<TransactionReceipt, FetchError | TxError>> {
+    const wallet = await this.getWallet(walletAddress)
+    const nonce = await wallet.nonce()
+    const attestations = await this.contractKit.contracts.getAttestations()
+
+    const approveTx = await attestations.approveAttestationFee(attestationsRequested)
+    const approveTxSig = await wallet.signMetaTransaction(approveTx.txo, nonce)
```

Comment:
nit: it might be better to increment nonce `nonce++` here in case we add more txs in the future since we won't need to worry about ordering.